src/Features/CSharp/Portable/CodeRefactorings/ConvertToRecord/CSharpConvertToRecordRefactoringProvider.cs

```diff
@@ -0,0 +1,623 @@
+
+                            var modifiedConstructor = constructor.WithInitializer(SyntaxFactory.ConstructorInitializer(SyntaxKind.ThisConstructorInitializer, thisArgs));
+
+                            modifiedMembers.Add(modifiedConstructor);
+                            continue;
+                        }
+
+                    case OperatorDeclarationSyntax op:
+                        {
+                            // keep track of equality overloads so we can potentially remove them later
+                            var opKind = op.OperatorToken.Kind();
+                            if (opKind == SyntaxKind.EqualsEqualsToken)
+                            {
+                                equals = op;
+                            }
+                            else if (opKind == SyntaxKind.ExclamationEqualsToken)
+                            {
+                                notEquals = op;
+                            }
+
+                            break;
+                        }
+
+                    case MethodDeclarationSyntax method:
+                        {
+                            // TODO: Ensure the single param is a StringBuilder
+                            if (method.Identifier.Text == "PrintMembers" &&
+                                method.ReturnType == SyntaxFactory.PredefinedType(SyntaxFactory.Token(SyntaxKind.BoolKeyword)) &&
+                                method.ParameterList.Parameters.IsSingle())
+                            {
+                                if (type.TypeKind == TypeKind.Class && !type.IsSealed)
+                                {
+                                    // ensure virtual and protected modifiers
+                                    modifiedMembers.Add(method.WithModifiers(SyntaxFactory.TokenList(
+                                        SyntaxFactory.Token(SyntaxKind.ProtectedKeyword),
+                                        SyntaxFactory.Token(SyntaxKind.VirtualKeyword))));
+                                }
+                                else
+                                {
+                                    // ensure private member
+                                    modifiedMembers.Add(method.WithModifiers(SyntaxFactory.TokenList(SyntaxFactory.Token(SyntaxKind.PrivateKeyword))));
+                                }
+
+                                continue;
+                            }
+
+                            break;
+                        }
+                }
+
+                // any other members we didn't change or modify we just keep
+                modifiedMembers.Add(member);
+            }
+
+            if (equals != null && notEquals != null)
+            {
+                if (ContainsEqualsInvocation(equals, type, semanticModel, cancellationToken))
+                {
+                    // we'll just assume that the != version always calls the == version or would be equivalent to !this.Equals(other)
```

Comment:
It's a little odd to be super opinionated about the equals method, but then just assume the best about the not equals method. It would make more sense to me to be either opinionted, or not care, about both.